src/metabase/driver.clj

```diff
@@ -398,6 +398,21 @@
     ;; Does the driver support percentile calculations (including median)
     :percentile-aggregations})
 
+(defmulti database-supports?
+  "Does this driver and specific instance of a database support a certain `feature`?
+  (A feature is a keyword, and can be any of the ones listed above in `driver-features`.
+  Note that it's the same set of `driver-features` with respect to
+  both database-supports? and supports?)
+
+  (supports? :mongo :set-timezone mongo-db) ; -> true"
```

Comment:
The docstrings (can) get formatted as Markdown so you should indent code examples 4 spaces

Reply:
https://guide.clojure.style/#markdown-docstrings

Using `[[...]]` for references too for stuff like `supports?` is definitely something we should do more of in the future... `cljdoc` supports it IIRC and CIDER definitely does. Not sure about other editors

https://guide.clojure.style/#document-references

One last nitpick: your example has `supports?` but the method is `database-supports?`

Reply:
I know the docstring for `supports?` is a little sparse, but I think we can take this opportunity to expand on the purpose of this method a bit while we're introducing it. It's nice that the docstring tells you _what_ it's supposed to do, but it doesn't really explain anything about what the purpose of this method is -- it would be nice to mention that it's used to feature-flag different things in the UI and QP (as well as tests). It would be nice too to suggest how a driver author might go about implementing it. e.g. something like

> Most drivers can always return `true` or always return `false` for a given feature (e.g., `:left-join` is not supported by _any_ version of Mongo DB). In some cases, a feature may only be supported by certain versions of the database engine; in this case, you can check the [database version column(s) where we record this info during sync] and determine whether a feature is supported for this particular database.

Driver methods should have a lot of documentation, since rando 3rd-party people are going to be implementing them